### ISSUE TITLE
If a download fails, retrying the installation also fails.

### ISSUE THREAD

**Issue by georgealegre** [kylef/swiftenv#112]
The temporary download files of Swift are stored in the `/tmp/` directory. If a download fails, these files aren't removed. Retrying the installation tries to continue the download from where it last failed but the swift.org server doesn't support resuming downloads so it fails.

```
$ swiftenv install https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
> Downloading https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
> /tmp/swiftenv-4.0.3-RELEASE-georgealegre ~
> ** Resuming transfer from byte position 167162654
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
```

**Comment by kylef** [comment]
That's strange, this is still working for me. It feature suggested in #53 and implemented in 33727158dc1551e9e34f51109f817f7c031822b7.

I wonder if there is some platform/curl version difference and possible bug elsewhere. Could you please confirm which version of curl you have installed along with the platform?

```shell
$ swiftenv install https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
Downloading https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
/var/folders/_l/ktm8nw2n7wg79yxf79_f58lr0000gn/T/swiftenv-4.0.3-RELEASE-kyle ~/Projects/tmp/test
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  159M    0  687k    0     0   197k      0  0:13:47  0:00:03  0:13:44  197k^C⏎                                                  

$ swiftenv install https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
Downloading https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz
/var/folders/_l/ktm8nw2n7wg79yxf79_f58lr0000gn/T/swiftenv-4.0.3-RELEASE-kyle ~/Projects/tmp/test
** Resuming transfer from byte position 1114112
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  158M    0  184k    0     0  67507      0  0:40:59  0:00:02  0:40:57 67484
```

(or hitting curl directly:)

```
$ curl  -C - -O https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  5  159M    5 9051k    0     0   959k      0  0:02:50  0:00:09  0:02:41 1549k^C⏎                                                  

$ curl  -C - -O https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz 

** Resuming transfer from byte position 9633792
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 20  150M   20 30.5M    0     0  1507k      0  0:01:42  0:00:20  0:01:22 2436k^C⏎                                                  

$ curl  -C - -O https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz 

** Resuming transfer from byte position 43122688
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  1  118M    1 1356k    0     0   275k      0  0:07:19  0:00:04  0:07:15  275k
```

**Comment by georgealegre** [comment]
Hi @kylef . I'm running Ubuntu 16.04 LTS 64 bit. My current version of Curl is:
`curl 7.47.0 (x86_64-pc-linux-gnu) libcurl/7.47.0 GnuTLS/3.4.10 zlib/1.2.8 libidn/1.32 librtmp/2.3`
However, I just retried this and it worked great. The problem is I ran `apt-get update`, `apt-get upgrade` and `apt-get dist-upgrade` a few hours ago, after reporting this issue. I bet I had an older version of Curl and that's why if failed. Although we aren't really reproducing a bug by hitting C-c in my opinion. I'm sorry I can't be of more help. I should have posted the version of Curl in the same description of the problem above.

**Comment by kylef** [comment]
@georgealegre Do you happen to recall why the first invocation of `swiftenv install` failed? I wonder if there is a specific error type that is unrecoverable.

I wonder if perhaps the file successfully downloaded, and something else caused the error. Then if swiftenv install is ran again, since the file is fully downloaded perhaps this error surfaces.

**Comment by kylef** [comment]
> I wonder if perhaps the file successfully downloaded, and something else caused the error. Then if swiftenv install is ran again, since the file is fully downloaded perhaps this error surfaces.

I can confirm this causes problems, I think to solve the problem. What we can do is download the file to a different file name (`$ORIGINAL_NAME.download`), once the download is complete we can move it over. I'll prepare a fix.

**Comment by georgealegre** [comment]
To be honest, I'm not completely sure. I ran the command and left the computer. What I can say for certain is that nothing more than what I posted was printed in my console. What I don't have is the error of the first invocation of the command. I just remember coming back to the computer and viewing what seemed like a download error (maybe a bad connection at my office). I immediately ran the command again and the error popped up.
